src/Knp/DictionaryBundle/Form/Type/DictionaryType.php

```diff
@@ -25,31 +25,29 @@ public function __construct(DictionaryRegistry $registry)
     /**
      * {@inheritdoc}
      */
-    public function setDefaultOptions(OptionsResolverInterface $resolver)
+    public function configureOptions(OptionsResolver $resolver)
     {
-        $choices = function (Options $options) {
+        $registry = $this->registry;
+
+        $choices = function (Options $options) use ($registry) {
             $name    = $options['name'];
-            $choices = $this->registry[$name]->getValues();
+            $choices = $registry[$name]->getValues();
 
             return $choices;
```

Comment:
I'm not sure how this can work properly. The doc for `ChoiceType` states:

> The choices option is an array, where the array key is the item's label and the array value is the item's value